components/nexus-core/src/main/java/org/sonatype/nexus/internal/blobstore/BlobStoreManagerImpl.java

```diff
@@ -245,22 +220,18 @@ private void untrack(final String name) {
   }
 
   /**
-   * Ensure that the path is not already used by another BlobStore
+   * Ensure that the configuration attributes are not already used by another BlobStore
    */
   private void validate(final BlobStoreConfiguration configuration) {
-    final String path = getPath(configuration.getAttributes());
-    boolean duplicatePath = Iterables.any(store.list(), new Predicate<BlobStoreConfiguration>()
-    {
-      @Override
-      public boolean apply(final BlobStoreConfiguration input) {
-        return path.equals(getPath(input.getAttributes()));
+    for (BlobStoreConfiguration blobStoreConfiguration : store.list()) {
+      if (Maps.difference(blobStoreConfiguration.getAttributes(), configuration.getAttributes()).areEqual()) {
```

Comment:
This looks suspicious to me - I think this test works now, but it seems coincidental to the fact that FileBlobStore only has one attribute, and it must have a unique value. This test seems to imply that it's okay to have two file blob stores writing to the same location, so long as they have some other setting that differs.